libfwupd/fwupd-enums.h

```diff
@@ -64,6 +64,28 @@ typedef enum {
 	FWUPD_TRUST_FLAG_LAST
 } FwupdTrustFlags;
 
+/**
+ * FwupdFeatureFlags:
+ * @FWUPD_FEATURE_FLAG_NONE:			No trust
+ * @FWUPD_FEATURE_FLAG_CAN_REPORT:		Can report successful update
+ * @FWUPD_FEATURE_FLAG_DETACH_TEXT:		Can show detach text
+ * @FWUPD_FEATURE_FLAG_DETACH_IMAGE:		Can show detach image
+ * @FWUPD_FEATURE_FLAG_UPDATE_TEXT:		Can show post-update text
+ * @FWUPD_FEATURE_FLAG_UPDATE_IMAGE:		Can show post-update image
```

Comment:
This has the potential to set up a weird dependency where you can only install certain updates using a GUI client and certain updates using a text client.  Maybe it can be avoided by the text client calls something like `jp2a`, but that's definitely not the ideal solution.

I think really you want the message to be that the client can show the user post-update steps and it's up to the firmware metainfo to specify both the text and graphical steps.

Reply:
> set up a weird dependency where you can only install certain updates using a GUI client

yes, that's kinda the idea :)

> I think really you want the message to be that the client can show the user post-update steps

Ahh, so we'd have something like FWUPD_FEATURE_FLAG_DETACH_ACTION and FWUPD_FEATURE_FLAG_UPDATE_ACTION and just silently fall back to the text if text+image was not available?

Reply:
>yes, that's kinda the idea :)

I find it impossible that there is a message that can ONLY be conveyed via GUI.  GUI is probably preferable, but making a hard requirement on a GUI is going to ruin a lot of use cases I feel like.

>Ahh, so we'd have something like FWUPD_FEATURE_FLAG_DETACH_ACTION and FWUPD_FEATURE_FLAG_UPDATE_ACTION and just silently fall back to the text if text+image was not available?

Yes!

Reply:
> Yes!

Will fix.